cpp/src/arrow/builder.cc

```diff
@@ -56,10 +76,46 @@ Status ArrayBuilder::Advance(int32_t elements) {
 
 Status ArrayBuilder::Reserve(int32_t elements) {
   if (length_ + elements > capacity_) {
+    // TODO(emkornfield) power of 2 growth is potentially suboptimal
```

Comment:
Yes this is what folly uses.  Seems like 1.5 edges out 2 on the very light survey done here: https://en.wikipedia.org/wiki/Dynamic_array#Growth_factor 

Ideally, we would benchmark once we have some real data in place.  In the absence of that 1.5 seems like a good default.